core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -989,13 +1009,13 @@ class ReplicaManager(val config: KafkaConfig,
     } else {
       // we can respond immediately
       val produceResponseStatus = initialProduceStatus.map { case (k, status) => k -> status.responseStatus }
-      responseCallback(produceResponseStatus)
+      responseCallback(produceResponseStatus.asJava)
```

Comment:
We use `initialProduceStatus` to build a Scala map and transfer to Java. Probably, we can build a Java map directly, so we can avoid `asJava` here.

```scala
      val produceResponseStatus = new util.HashMap[TopicIdPartition, PartitionResponse]()
      initialProduceStatus.foreach { case (k, status) => produceResponseStatus.put(k, status.responseStatus()) }
      responseCallback(produceResponseStatus)
```